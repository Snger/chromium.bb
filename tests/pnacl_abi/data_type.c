/*
 * tests whether our data type are lowered as expected (ILP32)
 * TODO: sizes of size_sb and size_ue are not quite right.
 */

#include <stdarg.h>
#include <setjmp.h>
#include <unwind.h>

int size_c = sizeof(char);
int size_i = sizeof(int);
int size_l = sizeof(long);

int size_p = sizeof(void*);

int size_f = sizeof(float);
int size_d = sizeof(double);

int size_vl = sizeof(va_list);
/* presumably a 5 word buffer */
/* c.f. http://llvm.org/docs/ExceptionHandling.html#llvm_eh_sjlj_setjmp */
int size_sb = sizeof(jmp_buf);
/* this should just be 4 words but we needed to hack it */
/* c.f. http://code.google.com/p/nativeclient/issues/detail?id=1107 */
int size_ue = sizeof(struct _Unwind_Exception);
