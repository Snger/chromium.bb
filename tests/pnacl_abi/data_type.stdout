target triple = "armv7-none-linux-gnueabi"

@size_c = unnamed_addr global i32 1
@size_i = unnamed_addr global i32 4
@size_l = unnamed_addr global i32 4
@size_p = unnamed_addr global i32 4
@size_f = unnamed_addr global i32 4
@size_d = unnamed_addr global i32 8
@size_vl = unnamed_addr global i32 16
@size_jb = unnamed_addr global i32 1024
@align_c = unnamed_addr global i32 1
@align_i = unnamed_addr global i32 4
@align_l = unnamed_addr global i32 4
@align_p = unnamed_addr global i32 4
@align_f = unnamed_addr global i32 4
@align_d = unnamed_addr global i32 8
@align_vl = unnamed_addr global i32 4
@align_jb = unnamed_addr global i32 8
